File: internal/commands/roadmap.go
package commands

import (
	"github.com/spf13/cobra"
	"github.com/trackfw/trackfw/internal/generators"
)

func newRoadmapCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "roadmap",
		Short: "Manage Roadmaps",
	}
	cmd.AddCommand(newRoadmapNewCmd(), newRoadmapMoveCmd())
	return cmd
}

func newRoadmapNewCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "new <title>",
		Short: "Create a new roadmap in backlog",
		Args:  cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			return generators.NewRoadmap(args[0])
		},
	}
}

func newRoadmapMoveCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "move <name> <state>",
		Short: "Move a roadmap between states (backlog|wip|blocked|done|abandoned)",
		Args:  cobra.ExactArgs(2),
		RunE: func(cmd *cobra.Command, args []string) error {
			return generators.MoveRoadmap(args[0], args[1])
		},
	}
}

File: npm/src/commands/roadmap.js
'use strict'
const { Command } = require('commander')
const { listRoadmaps, showRoadmap, moveRoadmap, newRoadmap } = require('../generators/roadmap')

const cmd = new Command('roadmap')
cmd.description('Manage Roadmaps')

cmd.command('new')
  .description('Create a new roadmap from a REQ')
  .option('-t, --title <title>', 'Roadmap title')
  .option('-r, --req <path>', 'Path to the linked REQ')
  .action(async (opts) => {
    const title = opts.title || 'New Roadmap'
    const reqPath = opts.req || ''
    newRoadmap(title, reqPath)
  })

cmd.command('list')
  .description('List all roadmaps grouped by state')
  .action(async () => {
    listRoadmaps()
  })

cmd.command('show <name>')
  .description('Show a roadmap by name (partial match)')
  .action(async (name) => {
    showRoadmap(name)
  })

cmd.command('move <name> <state>')
  .description('Move a roadmap between states (backlog|wip|blocked|done|abandoned)')
  .action(async (name, state) => {
    moveRoadmap(name, state)
  })

module.exports = cmd
